encode_utils/connection.py

```diff
@@ -165,7 +161,12 @@ def __init__(self, dcc_mode=None, dry_run=False, submission=False):
         self.post_logger = logging.getLogger(eu.POST_LOGGER_NAME)
         log_level = logging.INFO
         self.post_logger.setLevel(log_level)
-        self._add_file_handler(logger=self.post_logger, level=log_level, tag="posted")
+
+        # Add file handlers
+        if not no_log_file:
```

Comment:
Casey also made a branch trying to address the hardcoded PROFILES_URL issue (https://github.com/StanfordBioinformatics/encode_utils/compare/master...caseylitton:make-profiles-url-env-var), and for that, he used os.environ.get(). Does it also work to just add another arg to __init__ or do we need to have some environment variable set?

Reply:
Would it make ever make sense to have different URLs for the profile and for the `encode_utils.connection.Connection`? If not, then ideally we could point the profiles to the Connection's `dcc_url`, rather than needing to add another arg to the `__init__`.

I think the intention behind the way the Profile class (https://github.com/caseylitton/encode_utils/blob/aebb39ac4bc974d576fa74aefc0293f493b08044/encode_utils/profiles.py#L103) was written is to ensure that the `Profile` class itself (NOT the instances) takes care of getting all the schemas once, then instances can access the class-level variables to find the right schema.

It would be better to be able to use the`dcc_url` rather than needing to remember to set an env var. I can imagine a scenario where you have some script that connects to a demo, but you forgot to set the env var before importing encode_utils so the profiles are wrong (profiles are loaded at module import time, not a good engineering practice). 

However, this is a lot more work than Casey's fix. To get to the point where you could pass the url from the connection, you'd need to significantly refactor the profiles. It would make more sense to have just a single class called `Profiles` or something, that you could then obtain the correct instance of `Profile` for. Once at that point, the `Connection` can have its own instance of the `Profiles` class, with the profiles pulled from the `dcc_url`:
```
# Under the hood, sets self.profiles = Profiles(self.dcc_url)
conn = Connection("dcc_url")
# file_profile is of type Profile
file_profile = conn.profiles.get_profile_for_id(profile_id="/files/1/")
```

I'd support merging Casey's change, and revisit the profiles refactoring later.

Reply:
> Would it make ever make sense to have different URLs for the profile and for the encode_utils.connection.Connection? 
I can't really think of a case for this. If you ever tried to submit an object for a schema that doesn't match or doesn't exist on the Connection's DCC url, it would always be invalid (which is expected). Your suggestion to use dcc_url makes sense to me.

How would you want to handle this ticket for the log handling? Casey's ticket is separate, I brought it up since I'm not sure whether an arg is sufficient or if no_log_file also needs to be set as an environment variable.

Reply:
For this ticket I think an arg is fine.